### PR TITLE
[IP-1003]: Add extra field title

### PR DESCRIPTION
**Pull request by nielsdrost7**
- Add extra field title
- Simplification conditional
#1059 
#1003

### CODE REVIEW

File: application/modules/clients/controllers/Clients.php [hunk 1]
```diff
@@ -253,4 +267,12 @@ public function delete($client_id)
         $this->mdl_clients->delete($client_id);
         redirect('clients');
     }
+
+    private function get_client_title_choices(): array
+    {
+        return array_map(
```
Comment (nielsdrost7) [inline application/modules/clients/controllers/Clients.php:273]: @naui95 
so if we alllow fn functions, we cannot support PHP 7.4 anymore, correct?

@KevinJ30 I really like this function man, kudos!

File: application/modules/clients/models/Mdl_clients.php [hunk 2]
```diff
@@ -35,93 +35,97 @@ public function default_order_by()
 
     public function validation_rules()
     {
-        return array(
-            'client_name' => array(
+        return [
+            'client_title' => [
+                'field' => 'client_title',
+                'label' => trans('client_title'),
```
Comment (nielsdrost7) [inline application/modules/clients/models/Mdl_clients.php:41]: @KevinJ30 I removed the "required" part from the validation.
When you enter a new client the "title" isn't required

File: application/modules/clients/controllers/Clients.php (hunk 1 above)
Comment (naui95) [reply application/modules/clients/controllers/Clients.php:273]: @nielsdrost7 `fn` functions have been introduced in PHP 7.4 so we should be ok with them. Starting from IP1.6 we also state that minimum PHP version is 8.0 so this should not be an issue.

File: application/modules/clients/Enums/ClientTitleEnum.php [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+enum ClientTitleEnum: string
```
Comment (naui95) [inline application/modules/clients/Enums/ClientTitleEnum.php:3]: @nielsdrost7 the solution provided is cool, but an `enum` will break backward compatibility. Enums were introduced in PHP 8.1 (https://www.php.net/manual/en/language.types.enumerations.php). Can we find an alternative solution?

**Review by naui95 — CHANGES_REQUESTED** [review]

**Comment by nielsdrost7** [comment]
@KevinJ30 i moved the block for client_title to the bottom of the page (near Gender)
For the rest: awesome PR man!

**Review by naui95 — APPROVED** [review]

**Comment by nielsdrost7** [comment]
@KevinJ30 Merged, will be released in version 1.6.2

**Pull request merged**